keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1504 @@
+   // resources are local to nodes.
+   SelectedNode string
+
+   // When allocation is delayed, and the scheduler needs to
+   // decide on which node a Pod should run, it must first
+   // ask the plugin on which nodes the resource might be
+   // made available. To trigger that check, the scheduler must
+   // provide the names of nodes which might be suitable
+   // for the Pod. Will be updated periodically until
+   // the claim is allocated.
+   PotentialNodes []string
+
+   // A change of the node candidates triggers a check
+   // on which nodes the resource could be made available.
+   // This can change, so the plugin must refresh
+   // this information periodically until a node gets
+   // selected by the scheduler.
+   SuitableNodes []string
+
+   // An allocated resource is available on nodes that match this
+   // selector. If nil, the resource is available everywhere.
+   AvailableOnNodes *corev1.NodeSelector
+
+   // Arbitrary data returned by the plugin after a successful allocation.
+   // This data is passed to the plugin for all operations involving
+   // the allocated resource. This is opaque for Kubernetes.
+   // Plugin documentation may explain to users how to interpret
+   // this data if needed.
+   //
+   // The attributes must be sufficient to deallocate the resource
+   // because the ResourceClass might not be available anymore
+   // when deallocation starts.
+   Attributes map[string]string
+
+    // UsersLimit determines how many entities are allowed to use this resource
+    // at the same time. The default is 1. -1 enables the usage by an unlimited number
+   // of users. Individual containers in a pod are not counted as users, only the Pod
+   // is.
+   UserLimit int
+
+   // UsedBy indicates which entities are currently using the resource.
+   // Usually those are Pods. Only Pods listed as users can be scheduled,
+   // all others must wait. Updated by kube-scheduler as part of Pod scheduling
+   // (TBD – a separate controller might also work).
+   UsedBy []metav1.OwnerReference
+}
+
+type ResourceClaimPhase string
+
+const (
+    // The claim is waiting for allocation by the plugin.
+    //
+    // If it uses delayed allocation, SuitableNodes must have
+    // been set, otherwise the plugin cannot proceed with
+    // the allocation.
+    ResourceClaimPending ResourceClaimPhase = “Pending”
+
+    // Set by the plugin once the resource has been successfully
+    // allocated. The scheduler waits for all resources used by
+    // a Pod to be in this phase.
+    ResourceClaimAllocated ResourceClaimPhase = “Allocated”
+
+    // It can happen that a resource got allocated for a Pod and
+    // then the Pod cannot run on the nodes where the allocated
+    // resource is available. The scheduler detects this and
+    // then sets the “reallocate” phase to tell the plugin that it must
+    // free the resource. The plugin does that and moves it
+    // back to pending.
```

Comment:
It is not very clear here if  "pending" here corresponds to ResourceClaimPhase or pod moving back to the pending state. Please clarify.

Reply:
"it" = ResourceClaimPhase. Fixed.